src/DIRAC/FrameworkSystem/scripts/dirac_login.py

```diff
@@ -288,16 +294,29 @@ def loginWithCertificate(self):
 
         # Create local proxy with group
         self.outputFile = self.outputFile or Locations.getDefaultProxyLocation()
-        result = chain.generateProxyToFile(self.outputFile, int(self.lifetime or 12) * 3600, self.group)
+        parameters = (self.outputFile, int(self.lifetime or 12) * 3600, self.group)
+
+        # Add a VOMS extension if the group requires it
+        if (result := chain.generateProxyToFile(*parameters))["OK"] and (result := self.__enableCS())["OK"]:
```

Comment:
Isn't this equivalent to 
```suggestion
        if chain.generateProxyToFile(*parameters)["OK"] and self.__enableCS()["OK"]:
```
?

Reply:
in your case we not have `result` variable

we should handle `result` if it's not `OK`:
```python
       if not result["OK"]:
            return result
```